crates/ruff/src/rules/ruff/rules/explicit_f_string_type_conversion.rs

```diff
@@ -39,59 +47,124 @@ impl AlwaysAutofixableViolation for ExplicitFStringTypeConversion {
     }
 }
 
+fn fix_explicit_f_string_type_conversion(
+    expr: &Expr,
+    formatted_values: &[(usize, TextRange)],
+    locator: &Locator,
+    stylist: &Stylist,
+) -> Result<Fix> {
+    // Replace the call node with its argument and a conversion flag.
+    let range = expr.range();
+    let content = locator.slice(range);
+    let mut expression = match_expression(content)?;
+    let formatted_string = match_formatted_string(&mut expression)?;
+
+    for (index, _) in formatted_values {
+        let mut formatted_string_expression =
+            match_formatted_string_expression(&mut formatted_string.parts[*index])?;
+        let call = match_call(&mut formatted_string_expression.expression)?;
+        let name = match_name(&mut call.func)?;
+        match name.value {
+            "str" => {
+                formatted_string_expression.conversion = Some("s");
+            }
+            "repr" => {
+                formatted_string_expression.conversion = Some("r");
+            }
+            "ascii" => {
+                formatted_string_expression.conversion = Some("a");
+            }
+            _ => unreachable!(),
+        }
+        formatted_string_expression.expression = call.args[0].value.clone();
+    }
+
+    let mut state = CodegenState {
+        default_newline: &stylist.line_ending(),
+        default_indent: stylist.indentation(),
+        ..CodegenState::default()
+    };
+    expression.codegen(&mut state);
+
+    Ok(Fix::automatic(Edit::range_replacement(
```

Comment:
Is there a chance that this refactor changes the semantic of the code?

Reply:
It should not.
https://peps.python.org/pep-3101/#explicit-conversion-flag